File: core/src/main/java/com/satis/viewmodel/core/ObserverWrapper.java
package com.satis.viewmodel.core;


import androidx.lifecycle.Observer;

public class ObserverWrapper<T> implements Observer<T> {
    private BaseLiveData liveData;
    private Observer<? super T> mObserver;
    private int mLastVersion;

    public ObserverWrapper(BaseLiveData liveData,Observer<? super T> observer) {
        this.liveData = liveData;
        mLastVersion = liveData.mVersion;
        mObserver = observer;
    }

    @Override
    public void onChanged(T t) {
        //此处做拦截操作
        if (mLastVersion>=liveData.mVersion){
            return;
        }
        mObserver.onChanged(t);
    }
}

File: viewmodel-core/src/main/java/com/satis/viewmodel/core/ObserverWrapper.kt
package com.satis.viewmodel.core

import androidx.lifecycle.Observer
import com.satis.viewmodel.core.BaseLiveData
import com.satis.viewmodel.core.BaseObserver

class ObserverWrapper<T>(private val liveData: BaseLiveData<*>, observer: Observer<in T>) :
    Observer<T> {
    private val mObserver: Observer<in T> = observer
    private val mLastVersion: Int = liveData.mVersion
    override fun onChanged(t: T) {
        //此处做拦截操作
        if (mLastVersion >= liveData.mVersion) {
            return
        }
        mObserver.onChanged(t)
    }

}